api/v1alpha1/types_etcd_test.go

```diff
@@ -62,15 +64,15 @@ var _ = Describe("Etcd", func() {
 			created = getEtcd("foo", "default")
 
 			By("creating an API obj")
-			Expect(k8sClient.Create(context.TODO(), created)).To(Succeed())
+			Expect(k8sClient.Create(context.Background(), created)).To(Succeed())
```

Comment:
As we are refactoring, I think it's a good idea to use `context.WithTimeout()` and use a constant timeout value all through the code

Reply:
This is possible, although we will need to use different timeout values for each kind of request and resource. For instance, the time required to get a secret may be significantly lesser than the time required to create an `Etcd` resource.

Reply:
@abdasgupta we plan to do all test-specific refactoring as part of a separate PR. See https://github.com/gardener/etcd-druid/issues/527 for follow-up. So making further changes to the tests is out of scope for this PR. You may also check the `NOTE` from @unmarshall in this PR's description.

Reply:
If you do not mind, can we do this as part of another PR? We have already increased the scope of this PR by a lot (from where we started).